daemon/emer-persistent-cache.c

```diff
@@ -1086,6 +1104,28 @@ emer_persistent_cache_read (EmerPersistentCache *self,
       local_variants[i] = regularize_post_storage (curr_variant);
     }
 
+  if (corrupt_data)
+    {
+      /* Somehow the circular file contains corrupt data, meaning that we
+       * can't trust its contents anymore so we clean up and return an error. */
+      gsize j = 0;
+
+      while (j < num_elems)
+        g_bytes_unref (elems[j++]);
+      g_free (elems);
+
+      while (j < i)
+        g_variant_unref (local_variants[j++]);
```

Comment:
I think `j` should be initialized to `0` before this loop, since the variants to free in `local_variants` have index between `0` and `i - 1`